pkg/collector/corechecks/snmp/devicecheck/devicecheck.go

```diff
@@ -113,7 +117,20 @@ func (d *DeviceCheck) getValuesAndTags(staticTags []string) ([]string, *valuesto
 		}
 	}()
 
-	err := d.doAutodetectProfile(d.session)
+	// Get any value under 1.3 (iso.org) to check if the device is reachable
+	getNextValue, err := d.session.GetNext([]string{"1.3"})
+	if err != nil {
+		deviceReachable = false
+		checkErrors = append(checkErrors, fmt.Sprintf("check device reachable: failed: %s", err))
```

Comment:
Note for reviewers: It's a possibility to return here, but we might not want to take the risk to break existing setup if for some reasons GetNext can fail, but other following calls succeeds (even if this might be unlikely).

Reply:
Indeed I was thinking what about splitting this into a different function, like `checkIsReachable`. It would really really help to track this on our side to get notified if and when this happens. I know it's not possible to send events from the agent to the DD org logs, could we find a way to get this knowledge ? 

Could be a different state, "unreachable but with at list one metric". What do you think?

Reply:
Anyway I would move the `isReachable` logic out of the `getMetricsAndTags` logic

Reply:
> Could be a different state, "unreachable but with at list one metric". What do you think?

Hum, I think we already know that with the help of `snmp.submitted_metrics`.
But needs to look at both reachable status + `snmp.submitted_metrics`.

> Anyway I would move the isReachable logic out of the getMetricsAndTags logic

Agree it can be better. Let me have a look if I can improve/refactor the code.

Few constraints I have are that:
- I need to return errors to create the service check service checks
- I would like if possible only connect/close one session (this is done in getValuesAndTags)

Reply:
Is `snmp.submitted_metrics` sent to metadata? Here I was thinking of letting the NDM backend know about this "unreachable with metrics scenario" to be able to track it, as it is not expected

Reply:
> Is snmp.submitted_metrics sent to metadata? 

It's not, and we probably shouldn't since it's a data that have higher chance to change than other static metadata.


> Here I was thinking of letting the NDM backend know about this "unreachable with metrics scenario" to be able to track it, as it is not expected

I think we have enough data to **manually** check device reachability (NDM) + # of metrics (snmp.submitted_metrics).

Adding a new status (or other way to pass the info to NDM) can possibly make it easier. We can discuss about it.